tools/build-templated-pages/src/examples.rs

```diff
@@ -107,10 +107,10 @@ pub(crate) fn check(what_to_run: Command) {
 
     if what_to_run.contains(Command::UPDATE) {
         let categories = parse_categories();
-        let examples_by_category: HashMap<String, Category> = examples
+        let examples_by_category: HashMap<Box<str>, Category> = examples
             .into_iter()
-            .fold(HashMap::<String, Vec<Example>>::new(), |mut v, ex| {
-                v.entry(ex.category.clone()).or_default().push(ex);
+            .fold(HashMap::<Box<str>, Vec<Example>>::new(), |mut v, ex| {
+                v.entry(ex.category.as_str().into()).or_default().push(ex);
```

Comment:
```suggestion
                v.entry_ref(ex.category.as_str()).or_default().push(ex);
```
Might be able to avoid allocation here entirely via `entry_ref`.